From b66a454de0b57c9e0ff60dbbbf65c98394a6f371 Mon Sep 17 00:00:00 2001
From: Chris Lalancette <clalance@redhat.com>
Date: Thu, 16 Jun 2011 14:30:29 -0400
Subject: [PATCH 10/13] Bail out of GM_SAVE_INSTANCE_NAME if the job is HELD.

A missing break statement meant that we would still try to
do GM_SAVE_INSTANCE_NAME, even if the job was in HELD
or REMOVED.  Fix that here.

Signed-off-by: Chris Lalancette <clalance@redhat.com>
---
 src/condor_gridmanager/dcloudjob.cpp |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/condor_gridmanager/dcloudjob.cpp b/src/condor_gridmanager/dcloudjob.cpp
index b2be778..3953800 100644
--- a/src/condor_gridmanager/dcloudjob.cpp
+++ b/src/condor_gridmanager/dcloudjob.cpp
@@ -433,6 +433,7 @@ void DCloudJob::doEvaluateState()
 
 				if ( (condorState == REMOVED) || (condorState == HELD) ) {
 					gmState = GM_DELETE;
+					break;
 				}
 
 				// Once RequestSubmit() is called at least once, you must
-- 
1.7.4.4

